modules/serverless-logic-overview-timeouts.adoc

```diff
@@ -0,0 +1,205 @@
+// Module included in the following assemblies:
+// * about/serverless-logic-overview.adoc
+
+
+:_content-type: CONCEPT
+[id="serverless-logic-overview-timeouts_{context}"]
+= Timeouts
+
+OpenShift Serverless Logic defines several timeouts configurations that you can use to configure maximum times for the workflow execution in different scenarios. You can configure how long a workflow can wait for an event to arrive when it is in a given state or the maximum execution time for the workflow.
+
+Regardless of where it is defined, a timeout must be configured as an amount of time or duration, which starts when the referred workflow element becomes active. Timeouts use the `ISO 8601 data and time standard` to specify a duration of time and follow the format `PnDTnHnMn.nS`, with days considered to be exactly 24 hours. For example, `PT15M` configures 15 minutes, and `P2DT3H4M` defines 2 days, 3 hours, and 4 minutes.
+
+[NOTE]
+====
+Month-based timeouts like `P2M`, or period of two months, are not valid since the month duration might vary. In that case, use `PT60D` instead.
+====
+
+[id="workflow-timeout_{context}"]
+== Workflow timeout
+
+To configure the maximum amount of time that a workflow can be running before being canceled, you can use the workflow timeouts. Once canceled, the workflow is considered to be finished, and is not accessible through a GET request anymore. Therefore, it behaves as if the interrupt was `true` by default.
+
+Workflow timeouts are defined with the top-level `timeouts` property. It can have two types, `string` and `object`. The `string` type defines an URI that points to a JSON or YAML file containing the workflow timeout definitions. The `object` type, is used to define the timeout definitions inline. 
+
+For example, to cancel the workflow after an hour of execution, use the following configuration:
+
+.Example of workflow timeout
+[source,json]
+----
+  {
+  "id": "workflow_timeouts",
+  "version": "1.0",
+  "name": "Workflow Timeouts",
+  "description": "Simple workflow to show the workflowExecTimeout working",
+  "start": "PrintStartMessage",
+  "timeouts": {
+    "workflowExecTimeout": "PT1H"
+  } ...
+}
+----
+
+[id="event-timeout_{context}"]
+== Event timeout
+
+When you define a state in a workflow, you can use the `timeouts` property to configure the maximum time to complete this state. When that time is overdue, the state is considered timed-out, and the engine continues the execution from this state. The execution flow depends on the state type. For example, a transition to a next state might be executed. 
+
```

Comment:
In the original document https://sonataflow.org/serverlessworkflow/main/core/timeouts-support.html#callback-state

Inside the section 

[id="event-timeout_{context}"]
== Event timeout

We had the following subsections to make the distinction between states more clear.

**Callback state timeout**
**Switch state timeout**
**Event state timeout**

It looks like this sub-sections has gone now, and the separation between the "three sub-sections" is a bit difficult to follow"

Reply:
Technically I can not go beyond level 3. Let's see I have added that in the content.